File: CodeShellCore/Data/Lookups/RecursionModel.cs
using CodeShellCore.Data.Recursion;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CodeShellCore.Data.Lookups
{
    public class RecursionModel : IRecursiveModel
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<IRecursiveModel> Children { get; set; }
        
        public long? ParentId { get; set; }
        public string Chain { get; set; }
        public string NameChain { get; set; }
        public bool HasContents { get; set; }
        public int ContentCount { get; set; }
        public DateTime? CreatedOn { get; set; }
        public long? CreatedBy { get; set; }
    }
}

File: Configurator.UI/Core/codeshell/helpers/recursionModel.ts
export class RecursionModel {
    id: number=0;
    name: string="";
    children: any[] = [];
    editing: boolean = false;
    parentId: number | null = null;
    chain: string = "";
    nameChain: string = "";
    hasContents?: boolean;
    isExpanded: boolean = false;

    contentCount?: number;


    static FromDB(item: any, lst?: number[]): RecursionModel {
        let ret: RecursionModel = Object.assign(new RecursionModel, item);
        if (ret.children.length > 0) {
            for (var i in ret.children)
                ret.children[i] = this.FromDB(ret.children[i]);

        }
        return ret;
    }
}